pyteomics/proforma.py

```diff
@@ -1579,22 +1578,22 @@ class ChargeState(object):
     ----------
     charge : int
         The total charge state as a signed number.
-    adducts : list[str]
+    adducts : list[tuple[str, int]]
         Each charge carrier associated with the molecule.
     '''
     __slots__ = ("charge", "adducts")
 
     def __init__(self, charge, adducts=None):
         if adducts is None:
-            adducts = []
+            adducts = [('H+', charge)]
         self.charge = charge
         self.adducts = adducts
 
     def __str__(self):
         tokens = [str(self.charge)]
         if self.adducts:
             tokens.append("[")
-            tokens.append(','.join(str(adduct) for adduct in self.adducts))
+            tokens.append(','.join(f'{adduct[1]}{adduct[0]}' for adduct in self.adducts))
```

Comment:
This may need to be adjusted to deal with the default case where the charge carriers are just protons and we just show a number. For instance, `PEPTIDE/2` and `PEPTIDE/2[+2H+]` are equivalent.

The adduct is one of the things being updated in v2.1 because the v2 specification is vague about how you write the adduct and how much charge an adduct contributes. See https://docs.google.com/document/d/1kD_BPpJluRJ5a6nqzDVvqMzs17NnMuHx/edit#heading=h.jiwt4vipxzug

Reply:
They do produce identical output currently:

```python

In [5]: str(proforma.ProForma.parse('PEPTIDE/2').charge_state)
Out[5]: '2[2H+]'

In [6]: str(proforma.ProForma.parse('PEPTIDE/2[+2H+]').charge_state)
Out[6]: '2[2H+]'
```

The changes in this part seem quite drastic, basically the adduct notation is completely different in v2.1.

Reply:
Ah, I see what you mean, it didn't occur to me that the `__str__` is used in `to_proforma`. This raises the question: should `to_proforma` be able to produce different version of ProForma, or should we just update it whenever v2.1 is adopted?

Reply:
We should probably focus on returning what the user provides by spec version, or at the level of the peptidoform.